tests/execution/test-dedupe.md

```diff
@@ -0,0 +1,46 @@
+# testing dedupe functionality
+
+```graphql @config
+schema @server(port: 8000) {
+  query: Query
+}
+
+type Query {
+  posts: [Post] @http(url: "http://jsonplaceholder.typicode.com/posts?id=1", dedupe: true)
+}
+
+type Post {
+  id: Int
+  title: String
+  body: String
+  userId: Int!
```

Comment:
can we add more complex use case?
something like.

```graphql
type Query {
  posts: [Post] @http(url: "http://jsonplaceholder.typicode.com/posts?id=1", dedupe: true)
  user: [User] @http(url: "http://jsonplaceholder.typicode.com/users?id=1&id=2", dedupe: true)
}

type Post {
  id: Int
  title: String
  body: String
  userId: Int!
  user: User
    @http(
      url: "http://jsonplaceholder.typicode.com/users"
      query: [{key: "id", value: "{{.value.userId}}"}]
      batchKey: ["id"]
      dedupe: true
    )
}

type User {
  id: Int
  name: String
}
```

Reply:
ensure that `user` has expectedHits set 1 to make sure dedupe is working correctly.